libs/process/stub.go

```diff
@@ -0,0 +1,149 @@
+package process
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+var stubKey int
+
+// WithStub creates process stub for fast and flexible testing of subprocesses
+func WithStub(ctx context.Context) (context.Context, *processStub) {
+	stub := &processStub{responses: map[string]reponseStub{}}
+	ctx = context.WithValue(ctx, &stubKey, stub)
+	return ctx, stub
+}
+
+func runCmd(ctx context.Context, cmd *exec.Cmd) error {
+	stub, ok := ctx.Value(&stubKey).(*processStub)
+	if ok {
+		return stub.run(cmd)
+	}
+	return cmd.Run()
+}
+
+type reponseStub struct {
+	stdout string
+	stderr string
+	err    error
+}
+
+type processStub struct {
+	reponseStub
+	calls           []*exec.Cmd
+	defaultCallback func(*exec.Cmd) error
+	responses       map[string]reponseStub
+}
+
+func (s *processStub) WithDefaultOutput(output string) *processStub {
```

Comment:
Not symmetric with the per-command functions.

Please use `WithDefaultStdout` and `WithDefaultStderr`.